src/memo/memo.mli

```diff
@@ -1,9 +1,22 @@
 open! Stdune
 
+type 'a build
+
+module type Build = sig
```

Comment:
I got confused by the purpose of this module type. Is there more than one module with this signature?

Reply:
Is the purpose to reduce duplication in signatures of `Fiber` and `Build`?

Reply:
It's used in `File_tree.Dir.Make_map_reduce`, so that we can traverse the tree in the `Action_builder` monad. In the future, I expect that `Action_builder` (and the rule production monad) will have this signature as well.

Reply:
I see, got it.